README.md

```diff
@@ -25,23 +25,34 @@ The 2048 Game project is a Python implementation of the popular 2048 puzzle game
 - **Past Grid Tracking**: Keep track of the previous grid states using a stack.
 - **Score Tracking**: Each move the score get updated
 - **Game Over and Victory Detection**: Detect when the game is over or when the player wins.
+- **Ranking System**: Maintaining a ranking of 10 most high scores.
+- **Tree-Based Move Suggestions**: Provides approximate suggestions for optimal moves using a tree structure.
 
 ## Data Structures Used
 
 - **2D Array**: Representing the game grid.
 - **Stack** (linked list-based): Storing past grid states.
 - **Dictionary**: Managing key game features.
 - **Set**: Use a set to keep track of empty celleces (tiles with value 0)
+- **Tree**: Used in the tree-based move suggestion system.
 
 ## Algorithms
 
 - **Merge Algorithm**: Sum two cells adjacent in the same direction of movement.
 - **Random Number Generator Algorithm**: Generate random numbers in the grid.
 - **Move Algorithm**: Move the grid in the specified direction.
+- **Sorting Algorithms (QuickSort)**: Used in ranking system to sort the high scores.
 
 ## Code Structure Explanation
 
 The code is organized into several components to enhance readability and maintainability.
+main.py: The entry point of the game. It initializes the GameBoard and runs the main game loop.
```

Comment:
- put main in a new line? maybe like this is more clear?
- add: _main.py_ so like this is more visible
- the 'm' in main is capital 'M'